### PR TITLE
add salary changes page

### PR DESCRIPTION
**Pull request by lukasschreiber**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -7,7 +7,7 @@ Global / onChangedBuildSource := ReloadOnSourceChanges
 
 name                     := "Reform"
 ThisBuild / version      := "0.1.0-SNAPSHOT"
-ThisBuild / scalaVersion := "3.2.2"
+ThisBuild / scalaVersion := "3.2.1"
```
Comment (mohe2015) [inline build.sbt:10]: This breaks code coverage. What is the reason for downgrading?

File: js/src/main/scala/webapp/entity/UIAttribute.scala [hunk 2]
```diff
@@ -47,6 +72,77 @@ case class UIAttribute[EntityType, AttributeType](
...
+
+  def renderEdit(entityVar: Var[Option[EntityType]]) = {
+    entityVar.map {
+      _.map(entity => {
+        val attr = getter(entity)
+        td(
+          input(
+            tpe := "date",
+            minAttr := min,
+            value := attr.getAll.map(x => editConverter(x)).mkString("/"),
```
Comment (mohe2015) [inline js/src/main/scala/webapp/entity/UIAttribute.scala:99]: do conflicts work?

File: js/src/main/scala/webapp/entity/UIAttribute.scala [hunk 3]
```diff
@@ -47,6 +72,77 @@ case class UIAttribute[EntityType, AttributeType](
...
+  }
+}
+
+case class UISelectAttribute[EntityType, AttributeType](
+    override val getter: EntityType => Attribute[AttributeType],
+    override val setter: (EntityType, Attribute[AttributeType]) => EntityType,
+    override val readConverter: AttributeType => String,
+    override val writeConverter: String => AttributeType,
+    override val placeholder: String,
+    options: Signal[List[UIOption[Signal[String]]]],
```
Comment (mohe2015) [inline js/src/main/scala/webapp/entity/UIAttribute.scala:118]: I think we could remove the inner signal by flattening etc. Maybe I will look into that.

File: js/src/main/scala/webapp/pages/SalaryChangesPage.scala [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import webapp.Repositories
+import webapp.entity.*
+import rescala.default.*
+import webapp.utils.Date
+
+private val salaryChangeValue: UIAttribute[SalaryChange, Double] = UIAttribute(
+  _._value,
+  (p, a) => p.copy(_value = a),
+  readConverter = _.toString,
+  writeConverter = _.toDouble,
```
Comment (mohe2015) [inline js/src/main/scala/webapp/pages/SalaryChangesPage.scala:27]: We should change that to a money-safe datatype so fixed decimal type

File: build.sbt (hunk 1 above)
Comment (lukasschreiber) [reply build.sbt:10]: Metals does not support 3.2.2 therefore there are no interactive dev tools

Comment (lukasschreiber) [reply build.sbt:10]: https://discord.com/channels/1036673661844852756/1037079260046835802/1067408223562366976

File: js/src/main/scala/webapp/entity/UIAttribute.scala (hunk 2 above)
Comment (lukasschreiber) [reply js/src/main/scala/webapp/entity/UIAttribute.scala:99]: conflicts with what?

File: js/src/main/scala/webapp/entity/UIAttribute.scala (hunk 3 above)
Comment (lukasschreiber) [reply js/src/main/scala/webapp/entity/UIAttribute.scala:118]: ok

File: js/src/main/scala/webapp/pages/SalaryChangesPage.scala [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+)
+
+private val salaryChangeFromDate: UIDateAttribute[SalaryChange, Long] = UIDateAttribute(
+  _._fromDate,
+  (p, a) => p.copy(_fromDate = a),
+  readConverter = Date.epochDayToDate(_, "dd.MM.yyyy"),
+  editConverter = Date.epochDayToDate(_, "yyyy-MM-dd"),
+  writeConverter = Date.dateToEpochDay(_, "yyyy-MM-dd"),
+  placeholder = "From",
+  min = "2023-01-24",
```
Comment (mohe2015) [inline js/src/main/scala/webapp/pages/SalaryChangesPage.scala:51]: In our use case it should be possible to choose a data from the past

Comment (lukasschreiber) [reply js/src/main/scala/webapp/pages/SalaryChangesPage.scala:51]: Of course, was just to test the min thing

File: js/src/main/scala/webapp/entity/UIAttribute.scala (hunk 2 above)
Comment (mohe2015) [reply js/src/main/scala/webapp/entity/UIAttribute.scala:99]: If you edit the same field on both sides (so press edit on both, change values on both, press save on both) a confilcted value is shown and when you click edit again it breaks. This currently doesn't work for dates and number fields. Maybe we could redesign the ui, that it shows multiple date selects then or shows the other values in a text below the editing field

File: js/src/main/scala/webapp/entity/UIAttribute.scala (hunk 3 above)
Comment (mohe2015) [reply js/src/main/scala/webapp/entity/UIAttribute.scala:118]: we could also keep it, we don't know exactly what the advantages and disadvantages are except more complicated types everywhere.

Comment (mohe2015) [reply js/src/main/scala/webapp/entity/UIAttribute.scala:118]: Currently it should rerender more efficiently in theory

File: js/src/main/scala/webapp/entity/UIAttribute.scala (hunk 2 above)
Comment (lukasschreiber) [reply js/src/main/scala/webapp/entity/UIAttribute.scala:99]: Think that is out of scope for this US we should create a new US for conflict handling

File: js/src/main/scala/webapp/entity/UIAttribute.scala (hunk 3 above)
Comment (lukasschreiber) [reply js/src/main/scala/webapp/entity/UIAttribute.scala:118]: I think in this case it would be okay to keep it because it is quite simple...

**Review by mohe2015 — APPROVED** [review]

**Pull request merged**